### PR TITLE
Feature/drag and drop functionality

### PR DESCRIPTION
**Pull request by JaHajRF**
drag and drop, move to top functionalities added

### CODE REVIEW

File: packages/webapp/src/components/SongsList/SongsList.component.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import moveToTopIcon from '../../assets/playlist/SongsListComponent/move-song.svg'
+import vinylYellowIcon from '../../assets/playlist/SongsListComponent/vinyl-yellow.svg'
+import { reorderList } from '../../utils/reorderList'
+import { SearchedSong } from '../SearchedSong/SearchedSong.component'
+import { MoveSongIcon, StyledSongsList, UserPlayListElement, UserPlaylistTextElement } from './SongsList.styles'
+import { ISongsListProps } from './SongsList.types'
+
+export const SongsList: FC<ISongsListProps> = ({ playlist, deleteSongFromPlaylist, setPlaylist }) => {
+  const onDragEnd = (result: DropResult): undefined => {
```
Comment (kowalski) [inline packages/webapp/src/components/SongsList/SongsList.component.tsx:12]: correct way is:

```
  const onDragEnd = React.useCallback((result: DropResult): undefined => {
   if (!result.destination) return
   setPlaylist(playlist => reorderList(playlist, result.source.index, result.destination.index))
  }, [setPlaylist])
```

...
        <DragDropContext onDragEnd={onDragEnd}>
```

This last part - using arrow function inside JSX is a recurring pattern in your code. You have be aware that each time you do it, you make it impossible to React to figure out that the component's props haven't changed, therefore it doesn't need to re-render
```

File: packages/webapp/src/pages/Audius.tsx [hunk 2]
```diff
@@ -100,30 +90,11 @@ export const Audius: React.FC = (): JSX.Element => {
...
-              </button>
-              <button onClick={() => editSongOrder(track, Math.min(index + 1, playlist.length - 1))} type="button">
-                <strong> v </strong>
-              </button>
-            </div>
-          ))}
-      </div>
+      {playlist.length > 0 && (
+        <button onClick={() => dispatch(playerActions.initialPlay())} type="button">
+          Dispatch play song
```
Comment (kowalski) [inline packages/webapp/src/pages/Audius.tsx:95]: This should happen without click of the button right?
Basically if you take a DJ seat and there is no song currently playing you should `dispatch(playerActions.initialPlay())`

**Review by kowalski — CHANGES_REQUESTED** [review]

File: packages/webapp/src/pages/Audius.tsx (hunk 2 above)
Comment (JaHajRF) [reply packages/webapp/src/pages/Audius.tsx:95]: It is going to be addressed in future commits

**Review by kowalski — APPROVED** [review]

File: packages/webapp/src/components/SongsList/SongsList.component.tsx [hunk 3]
```diff
@@ -9,14 +9,17 @@ import { MoveSongIcon, StyledSongsList, UserPlayListElement, UserPlaylistTextEle
 import { ISongsListProps } from './SongsList.types'
 
 export const SongsList: FC<ISongsListProps> = ({ playlist, deleteSongFromPlaylist, setPlaylist }) => {
-  const onDragEnd = (result: DropResult): undefined => {
-    if (!result.destination) {
-      return
-    }
+  const onDragEnd = React.useCallback(
+    (result: DropResult): undefined => {
```
Comment (kowalski) [inline packages/webapp/src/components/SongsList/SongsList.component.tsx:13]: `void` not `undefined`

**Review by kowalski — APPROVED** [review]

**Pull request merged**